### ISSUE TITLE
Add LTS update

### ISSUE THREAD

**Issue by ineiti** [dedis/cothority#1536]
Currently the LTS is fixed and cannot be changed. Once we have the possibility to change the set of nodes in the roster, LTS should be able to change, too.

For this, we need dedis/kyber#326 and then add an api-call in the `calypso` service.

**Comment by kc1212** [comment]
For resharing the LTS, how are the conodes going to agree on the new roster? Leader could propose a new roster but the other nodes have no way to verify. Then the leader can just add anybody in the roster.

Is it reasonable to do something like this:
1. All conodes must obtain the new roster out-of-band, e.g., the administrator must add a new toml roster file to the config directory, or we can use the database. 
2. Roster starts the resharing protocol for the new roster.
3. All nodes verify that the roster is the same one as the one stored in config.
4. All nodes proceed to execute the protocol and creates new shares.

**Comment by ineiti** [comment]
With regard to the first one - this looks like an excellent use-case for something called a 'blockchain'. You create an `LTS`-contract and spawn an instance by giving it the roster and configuration of the nodes to be used in the sharing algorithm. Then this instance can be used to prove you have the right to do the DKG.

When you want to reshare the LTS, you update the instance (by respecting the DARC linked to it), then you get a proof for this instance and you can use it to do the reshare.

**Comment by ineiti** [comment]
With regard to Rabin / Pedersen - Nicolas told me already a long time ago I should switch, but never did it :( Is it complicated to change from one to another? As far as I understand, from an engineering point of view, there is no difference. Besides the fact that Pedersen has resharing...

So I'd say to create a Pedersen DKG and use that in `CreateLTS`.

**Comment by jeffallen** [comment]
I recommend against using this so-called 'blockchain'. I read on HN that it is a communist conspiracy and anyway it "doesn't scale".

**Comment by kc1212** [comment]
> With regard to the first one - this looks like an excellent use-case for something called a 'blockchain'. You create an `LTS`-contract and spawn an instance by giving it the roster and configuration of the nodes to be used in the sharing algorithm. Then this instance can be used to prove you have the right to do the DKG.
> 
> When you want to reshare the LTS, you update the instance (by respecting the DARC linked to it), then you get a proof for this instance and you can use it to do the reshare.

I was thinking that the secret-management cothority shouldn't need access to a blockchain system, at least that was the concept behind Calypso. But you're right that it's not easy to evolve the LTS group members without using byzcoin and Darc.

So, if we're going to use byzcoin anyway, do we need a dedicated LTS-contract that just stores the roster? Can we not just have one LTS per byzcoin. And the LTSID will have a field that identifies the relevant genesis block ID. If we do it this way, I think there's no need to have an extra contract. The roster for the LTS will follow the roster of byzcoin (otherwise you'll have to update both rosters). Then, every time the roster changes, calypso will start a re-sharing protocol.

**Comment by ineiti** [comment]
Hmm - I would like to keep the 'american dream' (no commies for you, jeff) of having the LTS and the ByzCoin separated. They have different roles, and in some setting it makes a lot of sense to have a big byzcoin roster, but a smaller LTS roster. So I'd say to go with the separated version. But you can talk to Lefteris to make sure.

**Comment by kc1212** [comment]
Right, so in this case, the LTS conodes might be be completely independent of the byzcoin conodes. It's totally up to the darc to decide how the LTS conodes evolve.

Which ever LTS conode is authorised to have a shared secret must act as a *client* of byzcoin and check the latest state of the `LTS`-contract instance, because these conodes might not have a byzcoin service on them.

**Comment by ineiti** [comment]
Exactly - I double-checked with Lefteris and he also said that the LTS-roster should be independent of the byzcoin roster.
